libqtile/backend/x11/xcbq.py

```diff
@@ -622,6 +624,10 @@ def configure(self, **kwargs):
         # since 1.12, uses switches to pack things sensibly
         if float(xcffib.__xcb_proto_version__) < 1.12:
             values = [i & 0xffffffff for i in values]
+        try:  # if we are a window, tell our group to move us (if StackMode is Above or Below)
+            self.group.windows.to(self, kwargs["stackmode"])
+        except (KeyError, AttributeError):
+            pass
```

Comment:
I just did it here because I wasn't sure this method wouldn't be called for things other than windows. Is that even a possibility?
I guess a type check would achieve the same goals without hiding exceptions.

Reply:
> PS: The general idea with this addition was that other places moving windows to "Above" or "Below" wouldn't circumvent the z-layering of the group.

Oh, right, then maybe we should rather modify the places where we used the `stackmode` argument of `configure` and replace these with a new function call like `window.above()`? That would make the code more straightforward, with less "spaghettis".

Reply:
That sounds reasonable. Accessing `StackMode` from various places does feel a little wrong anyway, that kind of stuff should be hidden away in my opinion. We do have a `cmd_bring_to_front` on the window already, although I think it makes normal windows float. Any idea why that is the case? Could I maybe remove the floating-part?

Reply:
The lines have been removed, and `StackMode` is now mostly set from within `Group`. `Window` has the new functions `to_relative_top` and `to_relative_bottom` (names very much discussable), which should be called when the StackMode needs to be changed on a window.